app/code/Magento/Elasticsearch7/etc/di.xml

```diff
@@ -221,6 +221,47 @@
                 <item name="elasticsearch7_copy_searchable_fields_to_search_field" xsi:type="object">Magento\Elasticsearch\Model\Adapter\FieldMapper\CopySearchableFieldsToSearchField</item>
                 <item name="elasticsearch7_add_default_search_field" xsi:type="object">Magento\Elasticsearch\Model\Adapter\FieldMapper\AddDefaultSearchField</item>
             </argument>
+            <argument name="dynamicTemplates" xsi:type="array">
```

Comment:
I would recommend create mapping params processors pull (Composite pattern) and pass params into it.
So any developer can add own processor that can change parameters as it wish.
Also there is problem with such structures you created that we cannot replace or delete items from it, only add new.

Reply:
Done